### PR TITLE
[Deepin Kernel SIG] [Debian] Apply some changes to better support package build system

### PR DESCRIPTION
**Pull request by Avenger-285714**
Those each patch needs to be evaluated before merging, some may require trimming or modification. 

The decision to merge a patch should be based on its merits.

Origin patches list:
```

# Changes to support package build system
debian/version.patch
debian/uname-version-timestamp.patch
debian/kernelvariables.patch
debian/mips-boston-disable-its.patch
debian/mips-ieee754-relaxed.patch
debian/mips-enable-r2-to-r6-emu-by-default.patch
debian/arch-sh4-fix-uimage-build.patch
debian/tools-perf-perf-read-vdso-in-libexec.patch
debian/tools-perf-install-python-bindings.patch
debian/wireless-add-debian-wireless-regdb-certificates.patch
debian/export-symbols-needed-by-android-drivers.patch
debian/android-enable-building-ashmem-and-binder-as-modules.patch
debian/documentation-drop-sphinx-version-check.patch
debian/perf-traceevent-support-asciidoctor-for-documentatio.patch
debian/kbuild-look-for-module.lds-under-arch-directory-too.patch
debian/kbuild-abort-build-if-subdirs-used.patch
debian/makefile-make-compiler-version-comparison-optional.patch
bugfix/all/revert-tools-build-clean-cflags-and-ldflags-for-fixdep.patch
debian/fixdep-allow-overriding-hostcc-and-hostld.patch
debian/linux-perf-remove-remaining-source-filenames-from-executable.patch
```

Link: https://salsa.debian.org/kernel-team/linux/-/tree/debian/latest/debian/patches?ref_type=heads

### CODE REVIEW

File: arch/mips/kernel/fpu-probe.c [hunk 1]
```diff
@@ -144,7 +144,12 @@ static void cpu_set_fpu_2008(struct cpuinfo_mips *c)
  * IEEE 754 conformance mode to use.  Affects the NaN encoding and the
  * ABS.fmt/NEG.fmt execution mode.
  */
-static enum { STRICT, EMULATED, LEGACY, STD2008, RELAXED } ieee754 = STRICT;
+enum ieee754_mode { STRICT, EMULATED, LEGACY, STD2008, RELAXED };
```
Comment (opsiff) [inline arch/mips/kernel/fpu-probe.c:147]: 相关提交已经合入
https://github.com/deepin-community/kernel/pull/528/commits/3d9826791b88e6abee19a09be4c97929fac35d7f

Comment (opsiff) [reply arch/mips/kernel/fpu-probe.c:147]: 不清楚这里需要选择 RELAXED 还是 EMULATED

Comment (Avenger-285714) [reply arch/mips/kernel/fpu-probe.c:147]: > 不清楚这里需要选择 RELAXED 还是 EMULATED

和EMULATED没关系，只是从STRICT改为RELAXED吧。

解释说明如下：
`The IEEE Std 754 compliance mode requested is determined by the main executable. Any dynamic shared objects loaded in addition shall respect the mode according to the rules set below.

For `strict' executables all the dynamic shared objects shall follow the same legacy-NaN or 2008-NaN ABI, as denoted by the EF_MIPS_NAN2008 flag described in Section 3.1. The value of the flag shall be the same across the executable and all the dynamic shared objects loaded together. Both `strict' and `legacy' dynamic shared objects shall be accepted, however `relaxed' ones shall be rejected regardless of the value of their EF_MIPS_NAN2008 flag.

For `relaxed' executables any dynamic shared objects shall be accepted, `strict', `relaxed' and `legacy' alike, regardless of the value of their EF_MIPS_NAN2008 flag.

For `legacy' executables the compliance mode is determined by the value of bit #25 in the `a_val' member of the AT_FLAGS entry of the auxiliary vector received from the kernel. The value of `0' shall make the dynamic loader follow the rules for `strict' executables. The value of `1' shall make the dynamic loader follow the rules for `relaxed' executables.`

Link: https://web.archive.org/web/20180830093617/https://dmz-portal.mips.com/wiki/MIPS_ABI_-_NaN_Interlinking

**Comment by Avenger-285714** [comment]
测试通过

**Pull request merged**